x-pack/plugins/reporting/server/config/create_config.test.ts

```diff
@@ -145,10 +140,49 @@ describe('Reporting server createConfig$', () => {
         encryptionKey: '888888888888888888888888888888888',
       })
     );
-    const mockConfig$: any = mockInitContext.config.create();
+    const mockConfig$ = createMockConfig(mockInitContext);
     const result = await createConfig$(mockCoreSetup, mockConfig$, mockLogger).toPromise();
 
     expect(result.capture.browser.chromium).toMatchObject({ disableSandbox: expect.any(Boolean) });
-    expect((mockLogger.warn as any).mock.calls.length).toBe(0);
+    expect(mockLogger.warn.mock.calls.length).toBe(0);
   });
+
+  for (const hostname of [
+    '0',
+    '0.0',
+    '0.0.0',
+    '0.0.0.0',
+    '0000:0000:0000:0000:0000:0000:0000:0000',
+    '::',
+  ]) {
+    it(`apply failover logic when hostname is given as ${hostname}`, async () => {
+      mockInitContext = coreMock.createPluginInitializerContext(
+        createMockConfigSchema({
+          encryptionKey: 'aaaaaaaaaaaaabbbbbbbbbbbbaaaaaaaaa',
+          // overwrite settings added by createMockConfigSchema and apply the default settings
+          // TODO make createMockConfigSchema _not_ default xpack.reporting.kibanaServer.hostname to 'localhost'
+          kibanaServer: {
+            hostname: undefined,
+            port: undefined,
+          },
+        })
+      );
+      mockCoreSetup.http.getServerInfo = jest.fn().mockImplementation(
+        (): HttpServerInfo => ({
+          name: 'cool server',
+          hostname,
+          port: 5601,
+          protocol: 'http',
+        })
+      );
+
+      const mockConfig$ = createMockConfig(mockInitContext);
+      const result = await createConfig$(mockCoreSetup, mockConfig$, mockLogger).toPromise();
+      expect(result.kibanaServer).toMatchObject({
+        hostname: 'localhost',
+        port: 5601,
+        protocol: 'http',
+      });
```

Comment:
I think it would be better to check that the promise is resolved and the result is not empty and has the property. In case when the test fails, jest can provide additional logging of where exactly it went wrong (e.g. promises rejects or empty result).

```suggestion
      await expect(
        createConfig$(mockCoreSetup, mockConfig$, mockLogger).toPromise()
      ).resolves.toHaveProperty(
        'kibanaServer',
        expect.objectContaining({
          hostname: 'localhost',
          port: 5601,
          protocol: 'http',
        })
      );
```